apm-lambda-extension/e2e-testing/e2e_test.go

```diff
@@ -0,0 +1,289 @@
+package e2e_testing
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"compress/zlib"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+var rebuildPtr = flag.Bool("rebuild", false, "rebuild lambda functions")
+var langPtr = flag.String("lang", "node", "the language of the Lambda test function : Java, Node, or Python")
+var timerPtr = flag.Int("timer", 20, "the timeout of the test lambda function")
+var javaAgentVerPtr = flag.String("java-agent-ver", "1.28.4", "the version of the java APM agent")
+
+func TestEndToEnd(t *testing.T) {
+
+	// Check the only mandatory environment variable
+	if err := godotenv.Load(".e2e_test_config"); err != nil {
+		log.Println("No additional .e2e_test_config file found")
+	}
+	if getEnvVarValueOrSetDefault("RUN_E2E_TESTS", "false") != "true" {
+		t.Skip("Skipping E2E tests. Please set the env. variable RUN_E2E_TESTS=true if you want to run them.")
+	}
+
+	supportedLanguages := []string{"node", "python", "java"}
+	if !isStringInSlice(*langPtr, supportedLanguages) {
+		processError(errors.New("unsupported language"))
+	}
+
+	samPath := "sam-" + *langPtr
+	samServiceName := "sam-testing-" + *langPtr
+
+	// Build and download required binaries (extension and Java agent)
+	buildExtensionBinaries()
+
+	// Java agent processing
+	if *langPtr == "java" {
+		if !folderExists(filepath.Join(samPath, "agent")) {
+			log.Println("Java agent not found ! Collecting archive from Github...")
+			retrieveJavaAgent(samPath, *javaAgentVerPtr)
+		}
+		changeJavaAgentPermissions(samPath)
+	}
+
+	// Initialize Mock APM Server
+	mockAPMServerLog := ""
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/intake/v2/events" {
+			bytesRes, _ := getDecompressedBytesFromRequest(r)
+			mockAPMServerLog += string(bytesRes)
```

Comment:
Added one to ease debugging in case we get more than one request in [b508016](https://github.com/elastic/apm-aws-lambda/pull/104/commits/b508016befaf8fc8b4e1901f8339ed038cf111b1)